src/python/pants/backend/go/goals/debug_goals.py

```diff
@@ -114,6 +121,74 @@ async def dump_go_import_paths_for_module(
     return DumpGoImportPathsForModule(exit_code=0)
 
 
+class GoExportCgoCodegenSubsystem(GoalSubsystem):
+    name = "go-export-cgo-codegen"
+    help = "Export files generated by Cgo."
+
+
+class GoExportCgoCodegen(Goal):
+    subsystem_cls = GoExportCgoCodegenSubsystem
+
+
+@goal_rule
+async def go_export_cgo_codegen(
+    targets: Targets,
+    console: Console,
+    distdir_path: DistDir,
+    workspace: Workspace,
+    golang_subsystem: GolangSubsystem,
+) -> GoExportCgoCodegen:
+    if not golang_subsystem.cgo_enabled:
+        raise ValueError("Nothing to export since cgo is disabled.")
+
+    go_package_targets = [tgt for tgt in targets if tgt.has_field(GoPackageSourcesField)]
+    for tgt in go_package_targets:
+        # Analyze the package and ensure it is actually contains cgo code.
+        analysis_wrapper = await Get(
+            FallibleFirstPartyPkgAnalysis, FirstPartyPkgAnalysisRequest(tgt.address)
+        )
+        if not analysis_wrapper.analysis:
+            console.write_stdout(
+                f"{tgt.address}: Failed to analyze target: {analysis_wrapper.stderr}\n"
+            )
+            continue
+        analysis = analysis_wrapper.analysis
+        if not analysis.cgo_files:
+            console.write_stdout(
+                f"{tgt.address}: Nothing to export because package does not contain Cgo files.\n"
```

Comment:
Under what conditions is this expected to happen? E.g., would this trigger if someone runs `./pants export src/go/::` and some packages don't contain cgo?

Reply:
That was my thinking. 

This is moot now, but also why write to stdout rather than logger.warn?

Reply:
I guess not moot since you "log" this way above as well

Reply:
I thought "normal" output from a goal should be output with `console.write_stdout`?